File: scraper_py/scrape_list.py
from playwright.sync_api import sync_playwright

BASE_URL = "https://tender-frontend-eight.vercel.app"


def get_tender_links():
    with sync_playwright() as p:
        browser = p.chromium.launch(
            headless=True,
            args=["--no-sandbox", "--disable-setuid-sandbox"]
        )
        page = browser.new_page()
        page.goto(BASE_URL, wait_until="networkidle")

        # wait for tender cards
        page.wait_for_selector(".card a", timeout=10000)

        anchors = page.query_selector_all(".card a")

        links = []
        for a in anchors:
            href = a.get_attribute("href")
            if href and href.startswith("/tender/"):
                links.append(BASE_URL + href)

        browser.close()

        # remove duplicates
        return list(set(links))

File: scrapper/scrapeList.js
import puppeteer from "puppeteer";

const BASE_URL = "https://tender-frontend-eight.vercel.app";

export async function getTenderLinks() {
  const browser = await puppeteer.launch({
    headless: true,
    args: ["--no-sandbox", "--disable-setuid-sandbox"]
  });

  const page = await browser.newPage();
  await page.goto(BASE_URL, { waitUntil: "networkidle2" });

  // wait for tender cards to appear
  await page.waitForSelector(".card a", { timeout: 10000 });

  const links = await page.$$eval(".card a", anchors =>
    anchors
      .map(a => a.getAttribute("href"))
      .filter(href => href && href.startsWith("/tender/"))
  );

  await browser.close();

  return [...new Set(links)].map(l => BASE_URL + l);
}